src/main/java/uk/gov/hmcts/cmc/claimstore/services/ClaimantResponseService.java

```diff
@@ -74,8 +78,26 @@ public void save(
         claimantResponseRule.assertCanBeRequested(claim, claimantId);
 
         Claim updatedClaim = caseRepository.saveClaimantResponse(claim, claimantResponse, authorization);
+
         claimantResponseRule.isValid(updatedClaim);
         formaliseResponseAcceptance(claimantResponse, updatedClaim, authorization);
+
+        Response response = claim.getResponse().orElseThrow(IllegalStateException::new);
+
+        if(isFullDefenseDisputeAcceptation(response, claimantResponse)) {
+            try {
+                appInsights.trackEvent(AppInsightsEvent.CLAIM_STAYED, REFERENCE_NUMBER, updatedClaim.getReferenceNumber());
+                caseRepository.saveCaseEvent(
+                    authorization,
+                    updatedClaim,
+                    CaseEvent.STAY_CLAIM
+                );
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Unknown case event provided in "
+                    + "'save' method for claim after claimant response ");
```

Comment:
Please include the original exception as the cause of this one

Reply:
Do you mean the log?